plugins/modules/swim_intent.py

```diff
@@ -1314,6 +1315,60 @@ def get_device_ip_from_id(self, device_id):
             self.log(error_message, "ERROR")
             raise Exception(error_message)
 
+    def check_swim_task_status(self, swim_task_dict, swim_task_name):
+        """
+        Check the status of the SWIM (Software Image Management) task for each device.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            swim_task_dict (dict): A dictionary containing the mapping of device IP address to the respective task ID.
+            swim_task_name (str): The name of the SWIM task being checked which is either Distribution or Activation.
+        Returns:
+            tuple: A tuple containing two elements:
+                - device_ips_list (list): A list of device IP addresses for which the SWIM task failed.
+                - device_count (int): The count of devices for which the SWIM task was successful.
+        Description:
+            This function iterates through the distribution_task_dict, which contains the mapping of
+            device IP address to their respective task ID. It checks the status of the SWIM task for each device by
+            repeatedly querying for task details until the task is either completed successfully or fails. If the task
+            is successful, the device count is incremented. If the task fails, an error message is logged, and the device
+            IP is appended to the device_ips_list and return a tuple containing the device_ips_list and device_count.
+        """
+
+        device_ips_list = []
+        device_count = 0
+
+        for device_ip, task_id in swim_task_dict.items():
+            start_time = time.time()
+            max_timeout = 1200
```

Comment:
1200 seconds we can make it as default value, but take input from the play..

Reply:
gave option in the playbook